Sources/Classes/RSDeviceModeManager.m

```diff
@@ -110,21 +116,23 @@ - (void) initiateCustomFactories {
 }
 
 - (void) replayMessageQueue {
-    @synchronized (self->eventReplayMessage) {
-        [RSLogger logDebug:@"RSDeviceModeManager: replayMessageQueue: replaying old messages with factory"];
-        if (self->eventReplayMessage.count > 0) {
-            NSMutableArray<NSNumber*>* rowIds = [self->eventReplayMessage.allKeys mutableCopy];
-            [RSUtils sortArray:rowIds inOrder:ASCENDING];
-            for(NSNumber* rowId in rowIds) {
-                [self makeFactoryDump:eventReplayMessage[rowId] FromHistory:YES withRowId:rowId];
-            }
+    RSDBMessage *dbMessage = [self->dbPersistentManager fetchAllEventsFromDBForMode:DEVICEMODE];
+    NSArray *messageIds = dbMessage.messageIds;
+    NSArray *messages = dbMessage.messages;
+    for (int i=0; i<messageIds.count; i++) {
+        id object = [RSUtils deSerializeJSONString:messages[i]];
+        NSNumber *rowId = [RSUtils convertStringIntoNSNumber:messageIds[i]];
```

Comment:
You can use 
`[messageIds[i] intValue]`

Reply:
Getting this warning: `Implicit conversion of 'int' to 'NSNumber *' is disallowed with ARC`, after using `[messageIds[i] intValue]`.